python/src/deltachat/chat.py

```diff
@@ -254,17 +254,19 @@ def send_image(self, path):
         return Message.from_db(self.account, sent_id)
 
     def prepare_message(self, msg):
-        """ create a new prepared message.
+        """ prepare a message for sending.
 
         :param msg: the message to be prepared.
-        :returns: :class:`deltachat.message.Message` instance.
+        :returns: a :class:`deltachat.message.Message` instance.
+           This is the same object that was passed in, which
+           has been modified with the new state of the core.
         """
         msg_id = lib.dc_prepare_msg(self.account._dc_context, self.id, msg._dc_msg)
         if msg_id == 0:
             raise ValueError("message could not be prepared")
-        # invalidate passed in message which is not safe to use anymore
-        msg._dc_msg = msg.id = None
-        return Message.from_db(self.account, msg_id)
+        # modify message in place to avoid bad state for the caller
+        msg._dc_msg = Message.from_db(self.account, msg_id)._dc_msg
```

Comment:
`Chat` modifying `Message` private property looks wrong, maybe this code should be moved to a new `Message` method that takes chat ID, so message prepares itself for the given chat ID. This method could be retained for backwards compatibility.

Just an idea, no need to change anything in this PR since it modified _dc_msg before (setting it to `None`) too.

Reply:
yes, modifying the Message private property is done twice in Chat already (in `send_msg()` and `send_prepared()`)  to update the message state, I basically copy+paste the comment etc.

btw, I don't know the use case of `prepare_message()`. I usually just create a message with `Message.new_empty()` and then use `Chat.send_msg()` what is `prepare_message()` for? for tests, to create the message without sending it?